receiver/k8sclusterreceiver/collection/pods_test.go

```diff
@@ -0,0 +1,223 @@
+// Copyright 2020, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package collection
+
+import (
+	"testing"
+
+	metricspb "github.com/census-instrumentation/opencensus-proto/gen-go/metrics/v1"
+	resourcepb "github.com/census-instrumentation/opencensus-proto/gen-go/resource/v1"
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/k8sclusterreceiver/testutils"
+)
+
+func TestPodAndContainerMetrics(t *testing.T) {
+	pod := newPodWithContainer("1")
+
+	actualResourceMetrics := getMetricsForPod(pod)
+
+	require.Equal(t, 2, len(actualResourceMetrics))
+
+	rm := actualResourceMetrics[0]
+
+	require.Equal(t, 1, len(actualResourceMetrics[0].metrics))
+	testutils.AssertResource(t, *rm.resource, k8sType,
+		map[string]string{
+			"k8s.pod.uid":        "test-pod-1-uid",
+			"k8s.pod.name":       "test-pod-1",
+			"k8s.node.name":      "test-node",
+			"k8s.namespace.name": "test-namespace",
+			"k8s.cluster.name":   "test-cluster",
+		},
+	)
+
+	testutils.AssertMetrics(t, *rm.metrics[0], "kubernetes/pod/phase",
+		metricspb.MetricDescriptor_GAUGE_INT64, 3)
+
+	rm = actualResourceMetrics[1]
+
+	require.Equal(t, 4, len(actualResourceMetrics[1].metrics))
+	testutils.AssertResource(t, *rm.resource, "container",
+		map[string]string{
+			"container.id":         "container-id",
+			"container.spec.name":  "container-name",
+			"container.image.name": "container-image-name",
+			"k8s.pod.uid":          "test-pod-1-uid",
```

Comment:
It is very unclear to me when do we expect to see uid vs name in the resource. Why do we not have a `k8s.node.uid` in this resource? May worth adding an issue to clarify when to use one or the other.

Reply:
Node information for a Pod is obtained from the Pod spec. In this case we don't add `k8s.node.uid` because it is not available in the Pod spec whereas the node name is available.

Reply:
Where can I find what is available or not in the Pod Specs?

Reply:
https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.18/#podspec-v1-core